libs/libcommon/src/libcommon/viewer_utils/asset.py

```diff
@@ -3,29 +3,40 @@
 
 import contextlib
 import os
+from functools import partial
 from os import makedirs
 from pathlib import Path
 from tempfile import NamedTemporaryFile
-from typing import Generator, List, Tuple, TypedDict
+from typing import Callable, Generator, List, Optional, TypedDict, Union, cast
+from uuid import uuid4
 
 from PIL import Image  # type: ignore
 from pydub import AudioSegment  # type:ignore
 
+from libcommon.constants import DATASET_SEPARATOR
+from libcommon.s3_client import S3Client
 from libcommon.storage import StrPath
+from libcommon.storage_options import DirectoryStorageOptions, S3StorageOptions
 
-DATASET_SEPARATOR = "--"
 ASSET_DIR_MODE = 0o755
 DATASETS_SERVER_MDATE_FILENAME = ".dss"
 SUPPORTED_AUDIO_EXTENSION_TO_MEDIA_TYPE = {".wav": "audio/wav", ".mp3": "audio/mpeg"}
 
 
-def create_asset_dir(
+def get_and_create_dir_path(
     dataset: str, config: str, split: str, row_idx: int, column: str, assets_directory: StrPath
-) -> Tuple[Path, str]:
+) -> Path:
     dir_path = Path(assets_directory).resolve() / dataset / DATASET_SEPARATOR / config / split / str(row_idx) / column
```

Comment:
why don't we use the url_dir_path (or the function get_url_dir_path)?

Reply:
Makes sense, done here https://github.com/huggingface/datasets-server/pull/1726/files#diff-4e03e1cb11067daafd8bc2c042d1e166ec0a88c4538a712e9757d125b5c6afbfR26